### PR TITLE
ROS2 Humble Updates

### PR DESCRIPTION
**Pull request by mr337**
This PR is an attempt to bring the ROS2 forked work back into the main repo. As mentioned https://github.com/nobleo/rviz_satellite/issues/99 to get the ball rolling. :smile: 

I attempted to do a rebase to get everything in sync from the great work from
* https://github.com/blacksoul000/rviz_satellite
* https://github.com/Kettenhoax/rviz_satellite

3.0.3 (2023-02-07)
------------------
* Fix `angles` dependency Cmake issue for proper compiling
* Update function to using updated due to deprecation

### CODE REVIEW

File: launch/publish_demo_data.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/python3
```
Comment (Timple) [inline launch/publish_demo_data.py:1]: This file doesn't seem to be installed

File: launch/publish_demo_data.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+
+class LooseFix():
+    """Set NavSatFix status to NO_FIX periodically"""
+    def __init__(self, phase, period):
+        self.phase = phase
+        self.period = period
+
+    def update(self, t, fix):
+        t_sec = t.nanoseconds / CONVERSION_CONSTANT
```
Comment (Timple) [inline launch/publish_demo_data.py:43]: ROS2 Time doesn't have a seconds property or function? Weird...

File: CMakeLists.txt [hunk 3]
```diff
@@ -22,78 +16,58 @@ find_package(rviz_default_plugins REQUIRED)
...
-)
-
-ament_export_include_directories(include)
-ament_target_dependencies(${PROJECT_NAME} rclcpp tf2_ros sensor_msgs rviz_common rviz_default_plugins)
-ament_export_dependencies(rclcpp tf2_ros sensor_msgs rviz_common rviz_default_plugins)
-ament_export_libraries(${PROJECT_NAME})
+target_include_directories(${PROJECT_NAME} PRIVATE src)
+target_link_libraries(${PROJECT_NAME} Qt5::Core Qt5::Gui Qt5::Network Qt5::Concurrent Qt5::Widgets)
 
 # prevent pluginlib from using boost
```
Comment (Timple) [inline CMakeLists.txt:39]: Is this still required as we do not depend on boost anymore?

File: README.md [hunk 4]
```diff
@@ -1,39 +1,28 @@
...
-## Demo
+Currently, the [OpenStreetMap](http://wiki.openstreetmap.org/wiki/Slippy_map_tilenames) convention for tile URLs is supported.
+This e.g. implies that only raster tiles (no vector tiles) are supported.
 
-The package contains a launch file for demonstration purposes.
-Use it to verify your installation and to get started:
+To try a demo, run the following commands to make Rviz circle around a geo point you define.
 
+```bash
+rviz2 -d $(ros2 pkg prefix --share rviz_satellite)/launch/demo.rviz
```
Comment (Timple) [inline README.md:18]: These files are not related to any launch.

I suggest to put the rviz config and the publish node in a `demo` folder instead of `launch`

File: launch/publish_demo_data.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    parser.add_argument('latitude', type=float)
+    parser.add_argument('longitude', type=float)
+    parser.add_argument('--noise', type=float)
+    parser.add_argument('--no-fix-phase', type=float)
+    parser.add_argument('--no-fix-period', default=1.0, type=float)
+    args = parser.parse_args()
+
+    rclpy.init()
+    node = rclpy.create_node('publish_demo_data')
+    publisher = node.create_publisher(NavSatFix, '/fix', 2)
```
Comment (Timple) [inline launch/publish_demo_data.py:62]: Topics should not be absolute. So `fix` instead of `/fix`.

Comment (Timple) [inline launch/publish_demo_data.py:62]: A queue of 2 is kind of strange. Latest message is the most relevant, so a queue of 1 suffices.

**Review by Timple — CHANGES_REQUESTED** [review]
Looking good! Thank you for taking the effort of combining these!

Also thanks to @blacksoul000 and @Kettenhoax.

Some small comments but since this branch works and the original one doesn't I'm inclined to be easy on this PR :slightly_smiling_face:

File: launch/publish_demo_data.py (hunk 2 above)
Comment (mr337) [reply launch/publish_demo_data.py:43]: I haven't seen it calculated that way. Most of the time I grab the first item in the tuple. But this does work.

```
In [8]: t
Out[8]: Time(nanoseconds=1676676204306391161, clock_type=ROS_TIME)

In [9]: t.seconds_nanoseconds()
Out[9]: (1676676204, 306391161)
```

File: CMakeLists.txt (hunk 3 above)
Comment (mr337) [reply CMakeLists.txt:39]: I am not exactly following the relationship between Qt5 and boost, think it is just over my head. I was able to trim the Qt5 deps to everything except Network.

File: README.md (hunk 4 above)
Comment (mr337) [reply README.md:18]: I went ahead and moved this to a `demo` dir and added the testing script to be installed so it can be ran via `ros2 run`.

File: launch/publish_demo_data.py (hunk 1 above)
Comment (mr337) [reply launch/publish_demo_data.py]: I had this file installed and moved to using `env python3`. Since humble is python3 I hope that isn't an issue.

**Comment by mr337** [comment]
I wish I had better answers for some of your questions. 

Here is proof this is up and running 
https://user-images.githubusercontent.com/413937/219820140-b216e87d-28ba-4e12-9c80-22c4f49b569d.mp4

**Comment by Timple** [comment]
Great!
I'll give it a run on my system and merge if all checks out!

**Review by Timple — APPROVED** [review]

**Comment by Timple** [comment]
Thank you!
